cmd2/cmd2.py

```diff
@@ -175,25 +175,30 @@ def cat_decorator(func):
     return cat_decorator
 
 
-def with_argument_list(func: Callable[[Statement], Optional[bool]],
-                       preserve_quotes: bool = False) -> Callable[[List], Optional[bool]]:
+def with_argument_list(*args: List[Callable], preserve_quotes: bool = False) -> Callable[[List], Optional[bool]]:
```

Comment:
I realize this works, but I'm confused. Why is args a list of Callables?

Reply:
Here I am making use of the ability to gather all positional arguments into a list, sort of like a varargs in C/C++ combined with the feature that any arguments after that are keyword-only arguments.

Reply:
Wow

Reply:
Here is a better and more complete explanation:  https://www.digitalocean.com/community/tutorials/how-to-use-args-and-kwargs-in-python-3